src/shogun2-core/shogun2-model/src/main/java/de/terrestris/shogun2/model/tree/TreeFolder.java

```diff
@@ -0,0 +1,168 @@
+package de.terrestris.shogun2.model.tree;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.persistence.Entity;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+import javax.persistence.ManyToOne;
+import javax.persistence.OrderColumn;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.hibernate.annotations.Cascade;
+import org.hibernate.annotations.CascadeType;
+
+import com.fasterxml.jackson.annotation.JsonIdentityInfo;
+import com.fasterxml.jackson.annotation.JsonIdentityReference;
+import com.fasterxml.jackson.annotation.ObjectIdGenerators;
+
+import de.terrestris.shogun2.model.layer.LayerGroup;
+
+/**
+ * The TreeFolder extends the TreeNode and contains n TreeNodes as children.
+ *
+ * @author Kai Volland
+ * @author Daniel Koch
+ *
+ */
+@Entity
+public class TreeFolder extends TreeNode {
+
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = 1L;
+
+	/**
+	 * 
+	 */
+	private boolean expanded;
+
+	/**
+	 * 
+	 */
+	@ManyToOne
+	@JsonIdentityInfo(
+			generator=ObjectIdGenerators.PropertyGenerator.class,
+			property="id"
+	)
+	@JsonIdentityReference(alwaysAsId=true)
+	private LayerGroup layerGroup;
```

Comment:
Why does the TreeLeaf has an AbstractLayer, but the TreeFolder a LayerGroup (which is also an AbstractLayer)? Can't we pull this up to the common parent class?